src/main/java/it/gov/pagopa/payhub/auth/service/a2a/registration/ClientSecretGeneratorService.java

```diff
@@ -0,0 +1,22 @@
+package it.gov.pagopa.payhub.auth.service.a2a.registration;
+
+import it.gov.pagopa.payhub.auth.service.DataCipherService;
+import org.springframework.stereotype.Service;
+
+import java.util.Base64;
+import java.util.function.Function;
+
+@Service
+public class ClientSecretGeneratorService implements Function<String, String> {
+
+	private final DataCipherService dataCipherService;
+
+	public ClientSecretGeneratorService(DataCipherService dataCipherService) {
+		this.dataCipherService = dataCipherService;
+	}
+
+	@Override
+	public String apply(String clientSecret) {
+		return Base64.getUrlEncoder().encodeToString(dataCipherService.hash(clientSecret));
```

Comment:
You have to encrypt the clientSecret, not hash!
We should be able to decrypt it in order to return to the client if requested!
Moreover, I think we could store the value directly as byte array... this way we will save some bit

Reply:
An other consideration! The encryption of the clientSecret is a detail of the DAO! not a detail of its generation logic!
It's wrong to use here the DataCipherService!
We should use the DataCipherService inside the ClientService in order to encrypt the clientSecret before to store it and decrypt before to return it to the requesting caller.

Having such a difference between the persisted and returned entity, it will make sense to uncouple the models and having a models: it will be the mapper to implement the encryption logic!
 (Client will have a clientSecretCipher field having type byte[], while the ClientDTO will have a String clientSecret)